sdk/src/wallet/operations/transaction/input_selection.rs

```diff
@@ -36,14 +38,27 @@ where
         mandatory_inputs: Option<HashSet<OutputId>>,
         remainder_address: Option<Address>,
         burn: Option<&Burn>,
+        context_inputs: Option<Vec<ContextInput>>,
+        issuer_id: Option<AccountId>,
         mana_allotments: Option<Vec<ManaAllotment>>,
+        payload: Option<TaggedDataPayload>,
     ) -> crate::wallet::Result<Selected> {
         log::debug!("[TRANSACTION] select_inputs");
         // Voting output needs to be requested before to prevent a deadlock
         #[cfg(feature = "participation")]
         let voting_output = self.get_voting_output().await?;
         let protocol_parameters = self.client().get_protocol_parameters().await?;
-        let slot_index = self.client().get_slot_index().await?;
+        let slot_commitment_id = self.client().get_issuance().await?.latest_commitment.id();
+        let reference_mana_cost = if let Some(issuer_id) = issuer_id {
+            Some(
+                self.client()
+                    .get_account_congestion(&issuer_id, None)
+                    .await?
+                    .reference_mana_cost,
+            )
+        } else {
+            None
```

Comment:
Don't we want first account_id as issuer_id if nothing else is provided? Otherwise it would not automatically allot in all the wallet tx methods?

Reply:
That decision should be made further up and passed in here

Reply:
Actually, maybe it's fine

Reply:
Yeah added another comment, I think we should do it further up where we have access to the tx options